src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -75,6 +75,7 @@ public boolean registerNewUser(UserRegistrationDto userRegistrationDto) {
     @Override
     public User loginUserWithOAuth2(DefaultOidcUser oauth2User) {
         var optionalUser = findByUsernameOrEmail(oauth2User.getEmail());
+        optionalUser.ifPresent(user -> userRepository.setOAuth2LoginToUserByEmail(oauth2User.getEmail()));
```

Comment:
You don't use **user** in the body of optional so just rename it to "**ignored**.

The new behaviour in **loginUserWithOAuth2** has to be covered at least with a unit test